reporter-web-app/src/sagas/index.js

```diff
@@ -26,14 +27,39 @@ export function* loadOrtResultData() {
 
     // Parse JSON report data embedded in HTML page
     const ortResultDataNode = document.querySelector('script[id="ort-report-data"]');
-    const ortResultDataTxt = ortResultDataNode ? ortResultDataNode.textContent : undefined;
-
-    if (!ortResultDataTxt || ortResultDataTxt.trim().length === 0) {
-        yield put({ type: 'APP::SHOW_NO_REPORT' });
-    } else {
-        const ortResultData = JSON.parse(ortResultDataTxt);
-        yield put({ type: 'APP::LOADING_ORT_RESULT_DATA_DONE', payload: ortResultData });
-        yield put({ type: 'APP::LOADING_PROCESS_ORT_RESULT_DATA_START' });
+
+    if (ortResultDataNode) {
+        const { type: dataType } = ortResultDataNode;
+        let ortResultData;
+        const ortReportDataTextContent = ortResultDataNode.textContent;
+
+        if (ortReportDataTextContent
+            && ortReportDataTextContent !== 'ORT_REPORT_DATA_PLACEHOLDER') {
+            if (dataType === 'application/gzip') {
+                // Decode base64 (convert ascii to binary)
+                const decodedBase64Data = atob(ortReportDataTextContent);
+
+                // Convert binary string to character-number array
```

Comment:
In general, comment sentences should have a dot at their end (also below).

Reply:
Fixed.